lib/mongodb.js

```diff
@@ -304,29 +304,40 @@ MongoDB.prototype.connect = function(callback) {
       err,
       client
     ) {
-      if (!err) {
+      if (err) {
+        if (callback) callback(err);
```

Comment:
I think `if (callback) callback(err);` should be after the debug log `if (self.debug) { }`